src/inmanta/deploy/state.py

```diff
@@ -194,13 +196,15 @@ def block_resource(self, resource: ResourceIdStr, details: ResourceDetails, is_t
         if resource in self.resource_state:
             if not is_transitive:
                 self.resource_state[resource].status = ComplianceStatus.UNDEFINED
-            self.resource_state[resource].blocked = BlockedStatus.YES
+                self.resource_state[resource].blocked = BlockedStatus.YES
```

Comment:
I don't think this is correct. Transient is not the same as transitive, and don't think we want to call this method at all for transiently blocked resources. We may want to rename it and/or clarify its docstring.

To clarify:
- transitive means we consider this resource blocked (`YES`) because it depends on another blocked (`YES`) resource. Somewhere at the head of that chain will be an `UNDEFINED` resource.
- transient means that the blocking might be short-lived, i.e. we can attempt to recover under certain conditions. This has nothing to do with `UNDEFINED`, and we do not transitively block resources because one of there dependencies is marked as transiently blocked (in a way we do, but we leave it up to the handler to drive the propagation).

Reply:
Reverted these changes